client/network/sync/src/lib.rs

```diff
@@ -1356,31 +1291,19 @@ where
 	}
 
 	fn send_block_request(&mut self, who: PeerId, request: BlockRequest<B>) {
-		let (tx, rx) = oneshot::channel();
-		let opaque_req = self.create_opaque_block_request(&request);
-
 		if self.peers.contains_key(&who) {
-			self.pending_responses
-				.insert(who, Box::pin(async move { (who, PeerRequest::Block(request), rx.await) }));
-		}
-
-		match self.encode_block_request(&opaque_req) {
-			Ok(data) => {
-				self.network_service.start_request(
-					who,
-					self.block_request_protocol_name.clone(),
-					data,
-					tx,
-					IfDisconnected::ImmediateError,
-				);
-			},
-			Err(err) => {
-				log::warn!(
-					target: "sync",
-					"Failed to encode block request {:?}: {:?}",
-					opaque_req, err
-				);
-			},
+			let network = self.network_service.clone();
```

Comment:
Why is a handle to `NetworkServiceProvider` not stored inside the `BlockDownloader`? Making a copy of each for every request doesn't seem right.

Making a copy of either one of them seems weird. Why can't `BlockDownloader` be some opaque service `ChainSync` calls in order to downloads blocks?

Reply:
Yes, changed the `BlockDownloader` to take the network handle during init time